src/sensors/tests/test_distant.py

```diff
@@ -0,0 +1,219 @@
+                       [1, 0, 0, 0],
+                       [0, 0, -1, 0],
+                       [0, 0, 0, 1]]
+    sensor = make_sensor(direction=[0, 0, -1])
+    assert ek.allclose(
+        sensor.world_transform().eval(0.).matrix,
+        world_reference
+    )
+
+    sensor = make_sensor(direction=[0, 0, -2])
+    assert ek.allclose(
+        sensor.world_transform().eval(0.).matrix,
+        world_reference
+    )
+
+
+@pytest.mark.parametrize("target", [
+    None,
+    [0.0, 0.0, 0.0],
+    [4.0, 1.0, 0.0],
+    [1.0, 1.0, 0.0],
+    [0.5, -0.3, 0.0],
+
+])
+@pytest.mark.parametrize("direction", [
+    [0.0, 0.0, 1.0],
+    [-1.0, -1.0, 0.0],
+    [2.0, 0.0, 0.0]
+])
+@pytest.mark.parametrize("ray_kind", ["regular", "differential"])
+def test_sample_ray(variant_scalar_rgb, direction, target, ray_kind):
+    sensor = make_sensor(direction, target)
+
+    for (sample1, sample2) in [[[0.32, 0.87], [0.16, 0.44]],
+                               [[0.17, 0.44], [0.22, 0.81]],
+                               [[0.12, 0.82], [0.99, 0.42]],
+                               [[0.72, 0.40], [0.01, 0.61]]]:
+
+        if ray_kind == "regular":
+            ray, _ = sensor.sample_ray(1., 1., sample1, sample2, True)
+        elif ray_kind == "differential":
+            ray, _ = sensor.sample_ray_differential(
+                1., 1., sample1, sample2, True)
+            assert not ray.has_differentials
+
+        # Check that ray direction is what is expected
+        assert ek.allclose(ray.d, ek.normalize(direction))
+
+        # Check that ray origin is outside of bounding sphere
+        # Bounding sphere is centered at world origin and has radius 1 without scene
+        assert ek.norm(ray.o) >= 1.
+
+
+def make_scene(direction=[0, 0, -1], target=None):
+    from mitsuba.core.xml import load_string
+
+    scene_xml = f"""
+    <scene version="2.0.0">
+        {xml_sensor(direction, target)}
+        <shape type="rectangle"/>
+    </scene>
+    """
+
+    return load_string(scene_xml)
+
+
+@pytest.mark.parametrize("target", [[0, 0, 0], [0.5, 0, 1]])
+def test_target(variant_scalar_rgb, target):
+    # Check if the sensor correctly targets the point it is supposed to
+    scene = make_scene(direction=[0, 0, -1], target=target)
+    sensor = scene.sensors()[0]
+    sampler = sensor.sampler()
+
+    ray, _ = sensor.sample_ray(
+        sampler.next_1d(),
+        sampler.next_1d(),
+        sampler.next_2d(),
+        sampler.next_2d()
+    )
+    si = scene.ray_intersect(ray)
+    assert si.is_valid()
+    assert ek.allclose(si.p, [target[0], target[1], 0.], atol=1e-6)
+
+
+@pytest.mark.parametrize("direction", [[0, 0, -1], [0.5, 0.5, -1]])
+def test_intersection(variant_scalar_rgb, direction):
+    # Check if the sensor correctly casts rays spread uniformly in the scene
+    direction = ek.normalize(direction)
+    scene = make_scene(direction=direction)
+    sensor = scene.sensors()[0]
+    sampler = sensor.sampler()
+
+    n_rays = 10000
```

Comment:
Could we use a smaller number here? Trying to keep the unit tests to run under 20 minutes ^^

Reply:
I reduced this loop to 1000 iterations, but the variance of results increased as a consequence.